Harvest.Web/ClientApp/src/Closeout/CloseoutContainer.tsx

```diff
@@ -0,0 +1,187 @@
+    .positive("Acreage Expense must be positive")
+});
+
+export const CloseoutContainer = () => {
+  const { projectId } = useParams<RouteParams>();
+  const [project, setProject] = useState<Project | undefined>();
+  const [newExpenseCount, setNewExpenseCount] = useState(0);
+  const [notification, setNotification] = usePromiseNotification();
+  const [finalAcreageExpense, setFinalAcreageExpense] = useState<FinalAcreageExpense>({ amount: 0 } as FinalAcreageExpense);
+  const history = useHistory();
+  const [beyondCloseoutDisplayDate, setBeyondCloseoutDisplayDate] = useState(false);
+  const [closeoutRequested, setCloseoutRequested] = useState(false);
+
+  const {
+    onChange,
+    InputErrorMessage,
+    getClassName,
+    onBlur,
+    formErrorCount,
+  } = useInputValidator<FinalAcreageExpense>(finalAcreageExpenseSchema);
+
+  useEffect(() => {
+    const cb = async () => {
+      const response = await fetch(`/Project/Get/${projectId}`);
+      if (response.ok) {
+        const proj: Project = await response.json();
+        setProject(proj);
+        setBeyondCloseoutDisplayDate(new Date().getTime() - new Date(proj.end).getTime() >= (7 * millisecondsPerDay));
+      }
+    };
+
+    cb();
+  }, [projectId]);
+
+  useEffect(() => {
+    if (!project) {
+      return;
+    }
+
+    if (finalAcreageExpense.amount === 0) {
+      setFinalAcreageExpense({ amount: roundToTwo(project.acres * (project.acreageRate.price / 12)) });
+    }
+  }, [project, finalAcreageExpense, setFinalAcreageExpense]);
+
+  useEffect(() => {
+    if (closeoutRequested) {
+      history.push(`/project/details/${projectId}`);
+    }
+  }, [closeoutRequested, history, projectId]);
+
+  const addAcreageExpense = async () => {
+    const request = fetch(`/Expense/CreateAcreage/${projectId}?amount=${finalAcreageExpense.amount}`, {
+      method: "POST"
+    });
+    setNotification(request, "Adding Acreage Expense", "Acreage Expense Added");
+    const response = await request;
+    if (response.ok) {
+      setNewExpenseCount(newExpenseCount + 1);
+    }
+  };
+
+  const [getConfirmation] = useConfirmationDialog({
+    title: "Closeout Project",
+    message: (
+      <ul>
+        <li>Generates a final invoice if there are any unbilled expenses</li>
+        <li>Sets project status to either Completed or CloseoutPending based on whether there are any pending invoices</li>
+      </ul>)
+  });
+
+  const closeoutProject = async () => {
+    if (!await getConfirmation()) {
+      return;
+    }
+
+    const request = fetch(`/Invoice/DoCloseout/${projectId}`, {
+      method: "POST"
+    });
+
+    setNotification(request, "Closing Out Project", async (response: Response) => {
+      const result = await response.json() as Result<number>;
+      setCloseoutRequested(true);
+      return result.message;
+    });
+  }
+
+  if (project === undefined) {
+    return <div>Loading ...</div>;
+  }
+
+  const canCloseout = project.status === "AwaitingCloseout" ||
```

Comment:
How does a project get this status in the first place?

Reply:
This is set automatically by the invoice service when the date is past project end.